gate-web/src/test/java/com/netflix/spinnaker/gate/security/oauth/OAuth2Test.java

```diff
@@ -51,4 +54,23 @@ void shouldRedirectOnOauth2Authentication() throws Exception {
 
     assertEquals(302, result.getResponse().getStatus());
   }
+
+  /** Test: Public endpoint should be accessible without authentication */
+  @Test
+  public void whenAccessingPublicEndpointThenSuccess() throws Exception {
+    mockMvc.perform(get("/auth/user")).andExpect(status().isOk());
+  }
+
+  /** Test: Secure endpoint should be accessible with OAuth2 authentication */
+  @Test
+  public void whenAuthenticatedWithOAuth2ThenAccessGranted() throws Exception {
+    mockMvc.perform(get("/credentials").with(oauth2Login())).andExpect(status().isOk());
+  }
+
+  /** Test: Logout should redirect to home */
+  @Test
+  @WithMockUser // Simulating an authenticated user
```

Comment:
This test passes for me even with `@WithMockUser` commented out.  Is there a stronger test possible?  Or maybe it's still a good test without it?

Reply:
https://github.com/spinnaker/gate/pull/1874/commits/233e3ea0c2b80abd98864a6aa6257e703f1fa3fe